specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2023-03-15-preview/managedCassandra.json

```diff
@@ -0,0 +1,2631 @@
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "cosmos-db.json#/definitions/CloudError"
+            }
+          },
+          "202": {
+            "description": "Accepted. Repair run show will be returned shortly.",
+            "headers": {
+              "azure-AsyncOperation": {
+                "type": "string",
+                "description": "URI to poll for completion status."
+              },
+              "location": {
+                "type": "string",
+                "description": "URI to poll for completion status."
+              }
+            }
+          },
+          "200": {
+            "description": "Success. Repair run show is returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/CassandraReaperRunStatus"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/cassandraClusters/{clusterName}/repairRuns/{repairRunId}/segments/{segmentId}/abort": {
+      "post": {
+        "operationId": "CassandraRepair_AbortSegment",
+        "x-ms-examples": {
+          "CosmosDBManagedCassandraClusterRepairAbort": {
+            "$ref": "./examples/CosmosDBManagedCassandraClusterRepairAbort.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/clusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/repairRunId"
+          },
+          {
+            "$ref": "#/parameters/segmentId"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "cosmos-db.json#/definitions/CloudError"
+            }
+          },
+          "202": {
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            },
+            "description": "Success. The async operation will begin as soon as possible."
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/cassandraClusters/{clusterName}/repairRuns/{repairRunId}/intensity/{intensityValue}": {
+      "post": {
```

Comment:
What is the purpose of this. This operation doesn't seem like a post action. Overall, this design needs to be thought through. There are way too many post actions. Can you model a new Resource Type, under which you can perform CRUD operations?

Reply:
This command POSTs an update to the repair, here "intensity" of the repair is being updated.

Reply:
Same comment as below.  This is not the correct number of URI segments for a POST action.  It can be "{repairRunId}/setIntensity" if needed

Reply:
@pilor our Control Plane changes are already deployed in all of prod. Can I please open an bug to track this to be fixed in the next release, since this is a preview swagger. At present we are not in a situation to risk this feature.

Reply:
This is a public API.  The fact that it has a preview suffix isn't relevant.  You won't be able to change this API once its released to customers without going through a multi-year breaking change process.  These URIs aren't valid and need to be fixed